waveorder/io/multipagetiff.py

```diff
@@ -247,10 +247,9 @@ def _create_position_array(self, pos):
                                                dtype=self.dtype)
         # add all the images with this specific dimension.  Will be blank images if dataset
         # is incomplete
-        for t in range(timepoints):
-            for c in range(channels):
-                for z in range(slices):
-                    self.position_arrays[pos][t, c, z, :, :] = self.get_image(pos, t, c, z)
+        for p, t, c, z in self.coord_map.keys():
```

Comment:
@ieivanov In the case of a z-stack not acquired for a specific channel, is the coordinate from the IndexMap metadata for this channel the middle z-slice?

i.e. If Pos0, Time0, Channel 1 does not have a z-stack, but the rest of the channels have 81 slices does the coordinate look like (P, T, C, Z) = (0, 0, 1, 40) in the metadata?

Reply:
That's correct. The way the zarr array is structured right now, you will get a zeros image if you ask for anything but the middle slice, which is the expected behavior.

Reply:
Cool, thanks for catching this!